internal/node/hybrid/certificate_validator.go

```diff
@@ -9,115 +9,182 @@ import (
 	"time"
 )
 
-type ValidationErrorType int
-
-const (
-	ErrorNoCert ValidationErrorType = iota
-	ErrorCertFile
-	ErrorReadFile
-	ErrorInvalidFormat
-	ErrorClockSkewDetected
-	ErrorExpired
-	ErrorParseCA
-	ErrorInvalidCA
-)
+type CertNotFoundError struct {
+	message string
+	cause   error
+}
```

Comment:
you don't need to repeat the same code for all error types


```suggestion
type CertNotFoundError struct {
	baseError
}

type CertFileError struct {
	baseError
}

type baseError struct {
    message string
    cause error
}

func (e *baseError) Error() string {
	if e.cause != nil {
		return fmt.Sprintf("%s: %v", e.message, e.cause)
	}
	return e.message
}

func (e *baseError) Unwrap() error {
	return e.cause
}
```